### PR TITLE
Consolidate `machine.go` duplicate code across hypervisors

### PR DESCRIPTION
**Pull request by jakecorrenti**
Consolidates duplicate code across the various hypervisors' `machine.go` file and places it either in the proper file that already exists or in the new `pkg/machine/machine_common.go` file.

[NO NEW TESTS NEEDED]

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by jakecorrenti** [comment]
I'm not a _huge_ fan of the way that `WriteConfig` and `SetRootful` are done, so I wouldn't be opposed to some ideas on how that could be done better.

`setupAPIForwarding` is also basically exactly the same across hypervisors, but is heavily dependent on package-specific implementations of functions (`dockerClaimSupported`, `alreadyLinked`, etc.), so I'm not sure of a good way to refactor that.

File: pkg/machine/machine_common.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+//go:build amd64 || arm64
+// +build amd64 arm64
+
+package machine
+
+import (
+	"os"
+)
+
+// getDevNullFiles returns pointers to Read-only and Write-only DevNull files
```
Comment (vrothberg) [inline pkg/machine/machine_common.go:16]: Non-blocking: s/getDevNullFiles/GetDevNullFiles/

File: pkg/machine/machine_common.go [hunk 2]
```diff
@@ -24,3 +27,30 @@ func GetDevNullFiles() (*os.File, *os.File, error) {
 
 	return dnr, dnw, nil
 }
+
+// AddSSHConnectionsToPodmanSocket adds SSH connections to the podman socket if
+// no ignition path is provided
+func AddSSHConnectionsToPodmanSocket(uid, port int, identityPath, name, remoteUsername string, opts InitOptions) error {
```
Comment (vrothberg) [inline pkg/machine/machine_common.go:36]: Nit: It's cleaner to return early and inverse the logic.
```golang
if len(opts.IgnitionPath) > 0 {
    fmt.Println("An ignition path was provided.  No SSH connection was added to Podman")
    return nil
}
// Rest of the code
```

It's a rule of thumb that in my experience makes code easier to maintain.

File: pkg/machine/machine_common.go [hunk 3]
```diff
@@ -54,3 +54,71 @@ func AddSSHConnectionsToPodmanSocket(uid, port int, identityPath, name, remoteUs
...
+// WaitAPIAndPrintInfo prints info about the machine and does a ping test on the
+// API socket
+func WaitAPIAndPrintInfo(forwardState APIForwardingState, name, helper, forwardSock string, noInfo, isIncompatible, rootful bool) {
+	suffix := ""
+	if name != DefaultMachineName {
+		suffix = " " + name
+	}
+
+	if isIncompatible {
+		fmt.Fprintf(os.Stderr, "\n!!! ACTION REQUIRED: INCOMPATIBLE MACHINE !!!\n")
```
Comment (vrothberg) [inline pkg/machine/machine_common.go:70]: Nit and not something to tackle in this PR.

golang supports raw strings where we can specify new lines which would be _much_ easier to read and maintian than printing each line separately as done below.

**Review by vrothberg — COMMENTED** [review]
LGTM
All very nice improvements!

The nits are mostly FYIs.

**Pull request merged**